import-scripts/fetch-dmp-data-for-import.sh

```diff
@@ -328,9 +325,9 @@ MY_FLOCK_FILEPATH="/data/portal-cron/cron-lock/fetch-dmp-data-for-import.lock"
 
     if [ $IMPORT_STATUS_HEME -eq 0 ] ; then
         # fetch new/updated heme samples using CVR Web service (must come after git fetching). Threshold is set to 50 since heme contains only 190 samples (07/12/2017)
-        waitOutDmpTumorServerInstabilityPeriod
+        drop_dead_instant_string=$(date --date="+2hours" -Iseconds) # 2 hours from now for both tumor and germline fetching
```

Comment:
Why is this capped to 2 hours and not 3 like the others - same question for a few instances below.

Reply:
The drop dead instant should be set to the point in the future where we want to abandon the current attempt to fetch samples from the cvr sample.

Looking at heme fetches at various points in the last month:
(tumor start)-->(tumor end) /// (germline start)-->(germline end)
23:11:04 --> 1:27:55 /// 2:30:09 --> 02:31:06
23:30:33 --> 1:49:40 /// 2:3023 --> 02:31:03
23:29:52 --> 2:10:41 /// 2:30:28 --> 2:31:09
23:36:05 --> 2:10:25 /// 2:30:09 --> 2:30:50
23:27:33 --> 2:23:12 /// 2:30:09 --> 2:30:51  (This was august 31 -> september 1)

The gap of time between the two fetches is for the current script code pause during the call to waitOutDmpTumorServerInstabilityPeriod

This shows that I was incorrect thinking that the heme fetch completed relatively quickly each night .. so thank you for pointing this out. The truth is quite the opposite --- the tumor sample fetches above all took more than 2 hours, and the last on was just short of 3 hours. This shows an urgency for us to address the single session per job execution problem with a maximum runtime of 3 hours.

It may not be as bad as it seems ... during a standard cvr job the order of operation is basically:
- get session id
- download the master list for the study
- fetch the json of all queued samples
- for each sample fetched in the previous step, fetch segment data for it (CVRVariantsProcessor)
- do all the processing of datatypes ... including annotation of variants
- requeue samples on the master list but which are not in our datafiles

So -- we basically have all the data before we start the mutation annotation step, which is the longest running step usually. For example, for the nearly 3 hour run on August 31, the mutation annotation started at 23:55:16 .. which is 28 minutes after startup. We will not need the session token again until the final (requeue) step, so if the token expires we will fail to requeue that night (maybe we will succeed the next night).

I suggest that we do prioritize the renewal of the session token when it is nearing expiration. But for the purposes of this PR I will make the drop_dead_instant be 3 hours after the start of the heme tumor fetch, and also I will have a separate drop_dead_instant for the germline fetch (perhaps giving that a 1 hour drop dead limit). Overall, we want to make sure that the dmp pipeline will fail and exit if the dmp server is not responsive at all. So I think I actually need to put some logic in the script to have a timestamp for the total process (of all cohorts) to complete. We do about 6 fetches, so if each waits for 3 hours to fail the whole process to final failure may take 18 hours, which is too much.

Reply:
The overall runtime for script fetch-dmp-impact-data is currently 11 hours (starting at 7pm and ending around 5:45am). I have imposed an overall limit of 12 hours for this script to complete. We also have logic in fetch-and-import-dmp-data-wrapper which skips over the import steps if it is after 7am anyway. The consume samples step later (during the import step) has been set to a 3 hour timeout. In the worst possible case, the dmp fetching and dmp importing is constrained to complete in 18 hours, so multi-day stalls should be avoided if the dmp server returns to normal service.